src/pages/Product.tsx

```diff
@@ -0,0 +1,57 @@
+import './../css/product.css'
+import { useParams } from 'react-router-dom'
+import useProductQuery from '../hooks/useProductQuery'
+
+type ProductPageProps = {
+  onAddToCart: (product: ProductItem) => void
+}
+
+function ProductPage({ onAddToCart }: ProductPageProps) {
+  //const productId = window.location.pathname.split('/product/')[1]
+  const { productId } = useParams<'productId'>()
+  const productQuery = productId ? useProductQuery(productId) : null
```

Comment:
We can never call hooks conditionally. React is letting this slip by because productId is always defined when this renders. However we can rewrite this to always call the hook by moving this conditional to occur inside of `useProductQuery`
```suggestion
  useProductQuery(productId)
```

Reply:
Ah I like this much better!